tests/test_her.py

```diff
@@ -323,6 +323,45 @@ def env_fn():
     model.learn(total_timesteps=100)
 
 
+def test_ep_length_after_loading_replay_buffer(tmp_path):
```

Comment:
I'd prefer a test dedicated to `truncate_last_trajectory`. In fact, the loading bug is simply a consequence of truncation not being performed correctly.
Something like: create a buffer, feed it with false interactions, truncate it, check that the warning is lifted (when it should be), feed it with a new interaction, and finally check that the attributes `observations`, `rewards`, ... are as expected.